catalog/nato/stanag4559/nato-stanag4559-source/src/main/java/org/codice/alliance/nato/stanag4559/source/Stanag4559Source.java

```diff
@@ -0,0 +1,942 @@
+        }
+        queryableAttributes = map;
+    }
+
+    /**
+     * Obtains the description of the source from the Library interface.
+     *
+     * @return a description of the source
+     */
+    private void setSourceDescription() {
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
+            LibraryDescription libraryDescription = library.get_library_description();
+            stringBuilder.append(libraryDescription.library_name + " : ");
+            stringBuilder.append(libraryDescription.library_description);
+        } catch (ProcessingFault | SystemFault e) {
+            LOGGER.error("{} : Unable to retrieve source description.", id, e);
+        }
+        String description = stringBuilder.toString();
+        if (StringUtils.isBlank(description)) {
+            LOGGER.warn("{} :  Unable to retrieve source description.", getId());
+        }
+        this.description = description;
+    }
+
+    public void destroy() {
+        orb.shutdown(true);
+        availabilityPollFuture.cancel(true);
+        scheduler.shutdownNow();
+    }
+
+    public void refresh(Map<String, Object> configuration) {
+        LOGGER.debug("Entering Refresh : {}", getId());
+
+        if (configuration == null || configuration.isEmpty()) {
+            LOGGER.error("{} {} : Received null or empty configuration during refresh.",
+                    this.getClass()
+                            .getSimpleName(),
+                    getId());
+            return;
+        }
+
+        String cxfUsername = (String) configuration.get(CXF_USERNAME);
+        if (StringUtils.isNotBlank(cxfUsername) && !cxfUsername.equals(this.cxfUsername)) {
+            setCxfUsername(cxfUsername);
+        }
+        String cxfPassword = (String) configuration.get(CXF_PASSWORD);
+        if (StringUtils.isNotBlank(cxfPassword) && !cxfPassword.equals(this.cxfPassword)) {
+            setCxfPassword(cxfPassword);
+        }
+        String id = (String) configuration.get(ID);
+        if (StringUtils.isNotBlank(id) && !id.equals(this.id)) {
+            setId(id);
+        }
+        String iorUrl = (String) configuration.get(IOR_URL);
+        if (StringUtils.isNotBlank(iorUrl) && !iorUrl.equals(this.iorUrl)) {
+            setIorUrl(iorUrl);
+        }
+        Integer pollInterval = (Integer) configuration.get(POLL_INTERVAL);
+        if (pollInterval != null && !pollInterval.equals(this.pollInterval)) {
+            setPollInterval(pollInterval);
+        }
+        Integer maxHitCount = (Integer) configuration.get(MAX_HIT_COUNT);
+        if (maxHitCount != null && !maxHitCount.equals(this.maxHitCount)) {
+            setMaxHitCount(maxHitCount);
+        }
+        init();
+    }
+
+    @Override
+    public String getDescription() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(describableProperties.getProperty(DESCRIPTION))
+                .append(System.getProperty("line.separator"))
```

Comment:
You can also do [`System.lineSeparator()`](http://docs.oracle.com/javase/7/docs/api/java/lang/System.html#lineSeparator())